packages/neon_framework/packages/account_repository/test/account_repository_test.dart

```diff
@@ -0,0 +1,564 @@
+// ignore_for_file: inference_failure_on_collection_literal
+
+import 'package:account_repository/account_repository.dart';
+import 'package:account_repository/src/utils/authentication_client.dart';
+import 'package:built_collection/built_collection.dart';
+import 'package:built_value_test/matcher.dart';
+import 'package:http/http.dart' as http;
+import 'package:mocktail/mocktail.dart';
+import 'package:neon_framework/testing.dart' show MockNeonStorage;
+import 'package:nextcloud/core.dart' as core;
+import 'package:nextcloud/nextcloud.dart';
+import 'package:nextcloud/provisioning_api.dart' as provisioning_api;
+import 'package:test/test.dart';
+
+class _FakeStatus extends Fake implements core.Status {}
+
+class _FakeUri extends Fake implements Uri {}
+
+class _FakeClient extends Fake implements http.Client {}
+
+class _FakePollRequest extends Fake implements core.ClientFlowLoginV2PollRequestApplicationJson {}
+
+class _FakeNextcloudClient extends Fake implements NextcloudClient {}
+
+class _DynamiteResponseMock<B, H> extends Mock implements DynamiteResponse<B, H> {}
+
+class _CoreClientMock extends Mock implements core.$Client {}
+
+class _AppPasswordClientMock extends Mock implements core.$AppPasswordClient {}
+
+class _ClientFlowLoginV2ClientMock extends Mock implements core.$ClientFlowLoginV2Client {}
+
+class _UsersClientMock extends Mock implements provisioning_api.$UsersClient {}
+
+class _AccountStorageMock extends Mock implements AccountStorage {}
+
+typedef _AccountStream = ({BuiltList<Account> accounts, String? active});
+
+void main() {
+  late AccountStorage storage;
+  late AccountRepository repository;
+
+  late core.$Client coreClient;
+  late core.$AppPasswordClient appPassword;
+  late core.$ClientFlowLoginV2Client clientFlowLoginV2;
+  late provisioning_api.$UsersClient users;
+
+  setUpAll(() {
+    registerFallbackValue(_FakeUri());
+    registerFallbackValue(_FakePollRequest());
+    MockNeonStorage();
+  });
+
+  setUp(() {
+    coreClient = _CoreClientMock();
+    appPassword = _AppPasswordClientMock();
+    clientFlowLoginV2 = _ClientFlowLoginV2ClientMock();
+    users = _UsersClientMock();
+
+    mockedClient = AuthenticationClient(
+      core: coreClient,
+      appPassword: appPassword,
+      clientFlowLoginV2: clientFlowLoginV2,
+      users: users,
+    );
+
+    storage = _AccountStorageMock();
+
+    repository = AccountRepository(
+      userAgent: 'userAgent',
+      httpClient: _FakeClient(),
+      storage: storage,
+    );
+  });
+
+  final accountList = BuiltList<Credentials>([
```

Comment:
```suggestion
  final credentialsList = BuiltList<Credentials>([
```
The name is a bit confusing throughout the tests